android/app/src/main/java/app/organicmaps/util/ThemeSwitcher.java

```diff
@@ -79,46 +47,36 @@ public void restart(boolean isRendererActive)
   {
     mRendererActive = isRendererActive;
     String theme = Config.getUiThemeSettings(mContext);
-    if (ThemeUtils.isAutoTheme(mContext, theme))
-    {
-      mAutoThemeChecker.run();
-      return;
-    }
+    System.out.println("theme:" + theme);
+    setAndroidTheme(theme);
 
-    UiThread.cancelDelayedTasks(mAutoThemeChecker);
-    setThemeAndMapStyle(theme);
+    String themeToApply = ThemeUtils.getAndroidTheme(mContext);
+    int style = getStyle(themeToApply);
+    System.out.println("themeToApply:" + themeToApply);
+    setThemeAndMapStyle(themeToApply, style);
+  }
+  
+  private void setAndroidTheme(@NonNull String theme) {
+    if (ThemeUtils.isFollowSystemTheme(mContext, theme)) {
+      AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM);
+    } else if (ThemeUtils.isNightTheme(mContext, theme)) {
+      AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES);
+    } else if (ThemeUtils.isDefaultTheme(mContext, theme)) {
+      AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
+    }
```

Comment:
Please follow existing code style.
```suggestion
  private void setAndroidTheme(@NonNull String theme)
  {
    if (ThemeUtils.isFollowSystemTheme(mContext, theme))
      AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM);
    else if (ThemeUtils.isNightTheme(mContext, theme))
      AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES);
    else if (ThemeUtils.isDefaultTheme(mContext, theme))
      AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
```